<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Biomarker Analysis Catalog - Dev - RNAG9</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../graphs/rnag10.html" rel="next">
<link href="../graphs/rnag08.html" rel="prev">
<link href="../assets/images/logo.png" rel="icon" type="image/png">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script type="text/plain" cookie-consent="tracking">

(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-125641273-1', 'auto');

ga('send', {
  hitType: 'pageview',
  'anonymizeIp': true,
});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en-US"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<link rel="stylesheet" href="../assets/css/style.scss">
<meta property="og:title" content="Biomarker Analysis Catalog - Dev - RNAG9">
<meta property="og:description" content="RNAseq Kaplan-Meier Graph">
<meta property="og:image" content="https://insightsengineering.github.io/biomarker-catalog/graphs/assets/images/logo.png">
<meta property="og:site_name" content="Biomarker Analysis Catalog - Dev">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Biomarker Analysis Catalog - Dev</span>
    </a>
  </div>
          <ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-dev" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Dev</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-dev">
<li>
    <a class="dropdown-item" href="https://insightsengineering.github.io/biomarker-catalog/stable">
 <span class="dropdown-text">Stable</span></a>
  </li>  
    </ul>
</li>
</ul>
<div class="quarto-navbar-tools tools-wide">
    <a href="https://app.slack.com/client/T028PB489D3/" title="Pharmaverse Slack" class="quarto-navigation-tool px-1" aria-label="Slack"><i class="bi bi-slack"></i></a>
    <a href="https://github.com/insightsengineering/biomarker-catalog" title="BAC GitHub" class="quarto-navigation-tool px-1" aria-label="Repository"><i class="bi bi-github"></i></a>
    <a href="https://insightsengineering.github.io/biomarker-catalog/index.xml" title="BAC RSS" class="quarto-navigation-tool px-1" aria-label="RSS"><i class="bi bi-rss"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
            <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../graphs/index.html">Graphs</a></li><li class="breadcrumb-item"><a href="../graphs/rnag01.html">RNAG</a></li><li class="breadcrumb-item"><a href="../graphs/rnag09.html">RNAG9</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../assets/images/logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Index</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tables</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/cpmt01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CPMT</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/cpmt01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CPMT1</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/CPMT2/cpmt02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CPMT2</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/CPMT2/cpmt02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CPMT2A</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/cpmt03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CPMT3</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/DT1/dt01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/DT1/dt01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/DT1/dt01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT1A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/DT1/dt01b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT1B</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/DT1/dt01c.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT1C</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/DT2/dt02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT2</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/DT2/dt02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT2A</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/TET1/tet01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">TET</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/TET1/tet01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">TET1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/TET1/tet01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">TET1A</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Graphs</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/ag01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">AG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/ag01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">AG1</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/DG1/dg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/DG1/dg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/DG1/dg01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG1A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/DG1/dg01b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG1B</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/dg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG2</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/DG3/dg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG3</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/DG3/dg03a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG3A</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/dg04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG4</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/KG1/kg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/KG1/kg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG1/kg01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG1A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG1/kg01b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG1B</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/KG2/kg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG2</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG2/kg02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG2A</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/kg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG3</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/KG4/kg04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG4</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG4/kg04a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG4A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG4/kg04b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG4B</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/KG5/kg05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG5</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG5/kg05a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG5A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/KG5/kg05b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG5B</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RFG1/rfg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RFG1/rfg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RFG1/rfg01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG1A</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RFG2/rfg02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG2</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RFG2/rfg02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG2A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RFG2/rfg02b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG2B</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RFG2/rfg02c.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG2C</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rfg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG3</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RG1/rg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RG1/rg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RG1/rg01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG1A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RG1/rg01b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG1B</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RG1/rg01c.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG1C</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RG2/rg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG2</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RG2/rg02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG2A</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/RG3/rg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG3</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RG3/rg03a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG3A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/RG3/rg03b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG3B</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/spg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SPG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/spg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SPG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/spg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SPG2</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/rnag01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG4</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag06.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG6</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag07.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG7</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag08.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG8</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag09.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">RNAG9</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG10</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/SFG1/sfg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth2 ">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/SFG1/sfg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG1</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG1/sfg01a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG1A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG1/sfg01b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG1B</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/SFG2/sfg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG2</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG2/sfg02a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG2A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG2/sfg02b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG2B</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG2/sfg02c.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG2C</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG2/sfg02d.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG2D</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/SFG3/sfg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG3</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG3/sfg03a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG3A</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG4</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/SFG5/sfg05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG5</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG5/sfg05a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG5A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG5/sfg05b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG5B</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG5/sfg05c.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG5C</span></a>
  </div>
</li>
      </ul>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/SFG6/sfg06a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG6</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="" class="collapse list-unstyled sidebar-section depth3 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG6/sfg06a.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG6A</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG6/sfg06b.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG6B</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/SFG6/sfg06c.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG6C</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content column-page-right" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../graphs/index.html">Graphs</a></li><li class="breadcrumb-item"><a href="../graphs/rnag01.html">RNAG</a></li><li class="breadcrumb-item"><a href="../graphs/rnag09.html">RNAG9</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">RNAG9</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
<p class="subtitle lead">RNAseq Kaplan-Meier Graph</p>
  <div class="quarto-categories">
    <div class="quarto-category">RNAG</div>
  </div>
  </div>



<div class="quarto-title-meta column-page-right">

    
  
    
  </div>
  


</header><hr>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">Setup: Import &amp; Data Manipulation</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Kaplan-Meier Graph</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">Teal Module for Kaplan-Meier Graph</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">Session Info</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<p>This page can be used as a template of how to produce Kaplan-Meier graphs for RNA-seq gene expression analysis using available <code>tern</code> and <code>hermes</code> functions, and to create an interactive Kaplan-Meier graph for RNA-seq gene expression analysis using <code>teal.modules.hermes</code>.</p>
<p>The code below needs both RNA-seq data (in <code>HermesData</code> format) and time-to-event data (in <code>ADTTE</code> format) as input.</p>
<p>We first prepare the time-to-event data. We define an event indicator variable, transform the time to months and filter down to the overall survival subset.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tern)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a>adtte_f <span class="ot">&lt;-</span> random.cdisc.data<span class="sc">::</span>cadtte <span class="sc">%&gt;%</span></span>
<span id="cb1-5"><a href="#cb1-5"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb1-6"><a href="#cb1-6"></a>    <span class="at">is_event =</span> .data<span class="sc">$</span>CNSR <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb1-7"><a href="#cb1-7"></a>    <span class="at">AVAL =</span> <span class="fu">day2month</span>(.data<span class="sc">$</span>AVAL),</span>
<span id="cb1-8"><a href="#cb1-8"></a>    <span class="at">AVALU =</span> <span class="st">"Months"</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb1-10"><a href="#cb1-10"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(.data<span class="sc">$</span>PARAMCD <span class="sc">==</span> <span class="st">"OS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then we prepare the RNA-seq data. See <a href="../graphs/rnag01.html">RNAG1</a> for basic details on how to import, filter and normalize <code>HermesData</code>. We use <code>col_data_with_genes()</code> to extract the sample variables (<code>colData</code>) from the object, together with a single specified gene or a specified gene signature. See <code><a href="https://insightsengineering.github.io/hermes/latest-tag/reference/gene_spec.html">?hermes::gene_spec</a></code> for details on how to do this. Then we use <code>inner_join_cdisc()</code> to join this genetic data with the <code>ADTTE</code> data from above. See the help page for more details, in particular how the join keys can be customized if needed - here we just join based on <code>USUBJID</code> by default.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">library</span>(hermes)</span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a>gene_data <span class="ot">&lt;-</span> <span class="fu">col_data_with_genes</span>(</span>
<span id="cb2-4"><a href="#cb2-4"></a>  <span class="at">object =</span> hermes_data,</span>
<span id="cb2-5"><a href="#cb2-5"></a>  <span class="at">assay_name =</span> <span class="st">"counts"</span>,</span>
<span id="cb2-6"><a href="#cb2-6"></a>  <span class="at">genes =</span> <span class="fu">gene_spec</span>(<span class="st">"GeneID:1820"</span>)</span>
<span id="cb2-7"><a href="#cb2-7"></a>)</span>
<span id="cb2-8"><a href="#cb2-8"></a>joined_data <span class="ot">&lt;-</span> <span class="fu">inner_join_cdisc</span>(</span>
<span id="cb2-9"><a href="#cb2-9"></a>  <span class="at">gene_data =</span> gene_data,</span>
<span id="cb2-10"><a href="#cb2-10"></a>  <span class="at">cdisc_data =</span> adtte_f</span>
<span id="cb2-11"><a href="#cb2-11"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can then cut the resulting gene column (we figure out the column name and save it in <code>arm_name</code> below) in the <code>joined_data</code> into quantile bins (in this example we want three equally sized groups).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>arm_name <span class="ot">&lt;-</span> <span class="fu">attr</span>(gene_data, <span class="st">"gene_cols"</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a>binned_data <span class="ot">&lt;-</span> joined_data <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-4"><a href="#cb3-4"></a>    <span class="at">gene_factor =</span> <span class="fu">cut_quantile_bins</span>(</span>
<span id="cb3-5"><a href="#cb3-5"></a>      joined_data[[arm_name]],</span>
<span id="cb3-6"><a href="#cb3-6"></a>      <span class="at">probs =</span> <span class="fu">c</span>(<span class="dv">1</span> <span class="sc">/</span> <span class="dv">3</span>, <span class="dv">2</span> <span class="sc">/</span> <span class="dv">3</span>)</span>
<span id="cb3-7"><a href="#cb3-7"></a>    )</span>
<span id="cb3-8"><a href="#cb3-8"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="#cb3-9"></a>  <span class="fu">df_explicit_na</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<p>It is now simple to create the Kaplan-Meier graph by providing the data set created above with the variable specification. Note that we specify the above created <code>gene_factor</code> as arm variable here.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>variables <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="at">tte =</span> <span class="st">"AVAL"</span>,</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="at">is_event =</span> <span class="st">"is_event"</span>,</span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="at">arm =</span> <span class="st">"gene_factor"</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>)</span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="fu">g_km</span>(</span>
<span id="cb4-7"><a href="#cb4-7"></a>  binned_data,</span>
<span id="cb4-8"><a href="#cb4-8"></a>  <span class="at">variables =</span> variables,</span>
<span id="cb4-9"><a href="#cb4-9"></a>  <span class="at">annot_coxph =</span> <span class="cn">TRUE</span></span>
<span id="cb4-10"><a href="#cb4-10"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="rnag09_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
<p>See <a href="../graphs/KG1/kg01.html">KG1</a> to <a href="../graphs/KG5/kg05.html">KG5</a> for additional customization options for the Kaplan-Meier graphs or the help page <code>?g_km()</code>.</p>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<p>We start by importing a <code>MultiAssayExperiment</code> and sample <code>ADTTE</code> data; here we use the example <code>multi_assay_experiment</code> available in <code>hermes</code> and example <code>ADTTE</code> data from <code>random.cdisc.data</code>. We can then use the provided teal module <code>tm_g_km</code> to include the corresponding interactive Kaplan-Meier analysis in our teal app. Note that by default the <code>counts</code> assay is excluded via the <code>exclude_assays</code> argument, but we can include it by just saying that we don’t want to exclude any assays. In case that we have different non-standard column names in our ADTTE data set we could also specify them via the <code>adtte_vars</code> argument, see the documentation <code>?tm_g_km</code> for more details.</p>
<div class="cell" data-webshot="webshot2" data-screenshot.force="true" data-screenshot.opts="{&quot;delay&quot;:15}">
<details class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">library</span>(teal.modules.hermes)</span>
<span id="cb5-2"><a href="#cb5-2"></a></span>
<span id="cb5-3"><a href="#cb5-3"></a>data <span class="ot">&lt;-</span> <span class="fu">teal_data</span>()</span>
<span id="cb5-4"><a href="#cb5-4"></a>data <span class="ot">&lt;-</span> <span class="fu">within</span>(data, {</span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="fu">library</span>(random.cdisc.data)</span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="fu">library</span>(dplyr)</span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="fu">library</span>(hermes)</span>
<span id="cb5-8"><a href="#cb5-8"></a>  MAE <span class="ot">&lt;-</span> multi_assay_experiment</span>
<span id="cb5-9"><a href="#cb5-9"></a>  ADTTE <span class="ot">&lt;-</span> random.cdisc.data<span class="sc">::</span>cadtte <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10"></a>    <span class="fu">mutate</span>(<span class="at">is_event =</span> .data<span class="sc">$</span>CNSR <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb5-11"><a href="#cb5-11"></a>})</span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="fu">datanames</span>(data) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"MAE"</span>, <span class="st">"ADTTE"</span>)</span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="fu">join_keys</span>(data)[<span class="st">"ADTTE"</span>, <span class="st">"ADTTE"</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"STUDYID"</span>, <span class="st">"USUBJID"</span>, <span class="st">"PARAMCD"</span>)</span>
<span id="cb5-14"><a href="#cb5-14"></a></span>
<span id="cb5-15"><a href="#cb5-15"></a>app <span class="ot">&lt;-</span> <span class="fu">init</span>(</span>
<span id="cb5-16"><a href="#cb5-16"></a>  <span class="at">data =</span> data,</span>
<span id="cb5-17"><a href="#cb5-17"></a>  <span class="at">modules =</span> <span class="fu">modules</span>(</span>
<span id="cb5-18"><a href="#cb5-18"></a>    <span class="fu">tm_g_km</span>(</span>
<span id="cb5-19"><a href="#cb5-19"></a>      <span class="at">label =</span> <span class="st">"Kaplan-Meier Graph"</span>,</span>
<span id="cb5-20"><a href="#cb5-20"></a>      <span class="at">adtte_name =</span> <span class="st">"ADTTE"</span>,</span>
<span id="cb5-21"><a href="#cb5-21"></a>      <span class="at">mae_name =</span> <span class="st">"MAE"</span>,</span>
<span id="cb5-22"><a href="#cb5-22"></a>      <span class="at">exclude_assays =</span> <span class="fu">character</span>()</span>
<span id="cb5-23"><a href="#cb5-23"></a>    )</span>
<span id="cb5-24"><a href="#cb5-24"></a>  )</span>
<span id="cb5-25"><a href="#cb5-25"></a>)</span>
<span id="cb5-26"><a href="#cb5-26"></a><span class="fu">shinyApp</span>(app<span class="sc">$</span>ui, app<span class="sc">$</span>server)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[INFO] 2024-05-22 17:30:41.1030 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1180 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1212 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1611 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1662 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1714 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1791 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1863 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.1971 pid:3767 token:[f2d44648] teal.slice Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.4947 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.4970 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.4983 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:41.4996 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5008 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5020 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5032 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5044 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5056 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5069 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5092 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:41.5105 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5118 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5130 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5142 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5154 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5166 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5179 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5191 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:41.5203 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5224 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:41.5236 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5251 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5265 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5277 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5289 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5301 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5313 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5325 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: NULL -&gt; 0
[INFO] 2024-05-22 17:30:41.5346 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5358 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5370 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5383 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5395 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5407 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5423 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:41.5443 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5455 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5487 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.5500 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:41.9072 pid:3767 token:[f2d44648] teal.modules.hermes Default Shiny inputs initialized: {"strata-sample_var":{},"simple_reporter-add_report_card_simple-add_report_card_button":0,"simple_reporter-download_button_simple-download_button":0,"simple_reporter-archive_load_simple-reporter_load":0,"simple_reporter-reset_button_simple-reset_reporter":0,"genes-select_none_button":0,"genes-select_all_button":0,"genes-text_button":0,"strata-levels_button":0,"plot-downbutton-downl":0,"plot-expand":0,"plot-expbut":0,"assay-name":"","genes-genes":{},"genes-fun_name":"Mean","adtte-paramcd":"","percentiles":[0,0.5],"plot-downbutton-file_format":"png","genes-lock_button":false,"plot-downbutton-file_name":"plot_20240522_173040"}
[INFO] 2024-05-22 17:30:41.9136 pid:3767 token:[f2d44648] teal.modules.hermes Default Shiny inputs initialized: {}
[INFO] 2024-05-22 17:30:41.9213 pid:3767 token:[f2d44648] teal.modules.hermes Default Shiny inputs initialized: {"name":""}
[INFO] 2024-05-22 17:30:41.9280 pid:3767 token:[f2d44648] teal.modules.hermes Default Shiny inputs initialized: {"select_none_button":0,"select_all_button":0,"text_button":0,"genes":{},"fun_name":"Mean","lock_button":false}
[INFO] 2024-05-22 17:30:41.9405 pid:3767 token:[f2d44648] teal.modules.hermes Default Shiny inputs initialized: {"sample_var":{},"levels_button":0}
[INFO] 2024-05-22 17:30:41.9521 pid:3767 token:[f2d44648] teal.modules.hermes Default Shiny inputs initialized: {"paramcd":""}
[INFO] 2024-05-22 17:30:42.0102 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.0119 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.0134 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0149 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0163 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0178 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0192 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0206 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0221 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0235 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0250 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0265 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0279 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0293 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.0308 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.0322 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.0336 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.0351 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: NULL -&gt; c(0, 0.5)
[INFO] 2024-05-22 17:30:42.0365 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: NULL -&gt; png
[INFO] 2024-05-22 17:30:42.0380 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.0394 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: NULL -&gt; plot_20240522_173040
[INFO] 2024-05-22 17:30:42.0490 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.0505 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0520 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0534 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0549 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0564 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0726 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0739 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0751 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0763 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0775 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0786 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0798 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.0809 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.0821 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.0833 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.0844 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: c(0, 0.5) -&gt; NULL
[INFO] 2024-05-22 17:30:42.0856 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: png -&gt; NULL
[INFO] 2024-05-22 17:30:42.0867 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.0879 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: plot_20240522_173040 -&gt; NULL
[INFO] 2024-05-22 17:30:42.0900 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.0940 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.0952 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0964 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0976 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.0987 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.0999 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.1010 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.1102 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.1119 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.1136 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.1153 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.1170 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.1187 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.1203 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.1220 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.1297 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.1310 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.1322 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.2310 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.2325 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2345 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2357 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.2369 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2381 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2393 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2405 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2418 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2430 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2441 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2461 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.2473 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2485 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2497 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2509 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2521 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2533 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2546 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2557 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.2569 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2590 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.2602 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2614 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2627 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2639 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2651 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2663 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2675 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2686 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2706 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2718 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2730 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2742 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2753 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2765 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2777 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.2797 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2809 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2841 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2853 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2873 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.2886 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2898 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2910 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2922 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2934 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2946 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2957 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2969 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2982 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.2994 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.3006 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.3017 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.3029 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3041 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.3053 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.3065 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: NULL -&gt; c(0, 0.5)
[INFO] 2024-05-22 17:30:42.3077 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: NULL -&gt; png
[INFO] 2024-05-22 17:30:42.3089 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.3101 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: NULL -&gt; plot_20240522_173040
[INFO] 2024-05-22 17:30:42.3112 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: NULL -&gt; 732
[INFO] 2024-05-22 17:30:42.3124 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: NULL -&gt; 1252.8
[INFO] 2024-05-22 17:30:42.3146 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3158 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3170 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3184 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3196 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3207 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3220 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3231 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3243 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3255 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3267 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3279 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3290 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.3302 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3314 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.3326 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.3338 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: c(0, 0.5) -&gt; NULL
[INFO] 2024-05-22 17:30:42.3350 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: png -&gt; NULL
[INFO] 2024-05-22 17:30:42.3362 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.3373 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: plot_20240522_173040 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3385 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: 732 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3397 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: 1252.8 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3419 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.3438 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.3450 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.3463 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.3475 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.3487 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3499 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.3510 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.3530 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3542 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3554 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3566 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3578 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.3590 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.3602 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3613 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.3635 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.3647 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.3659 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on paramcd: NULL -&gt; </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: 'experiments' dropped; see 'drops()'</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[INFO] 2024-05-22 17:30:42.5961 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.5975 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.5995 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6007 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.6020 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6032 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6043 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6055 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6067 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6080 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6092 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6112 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.6124 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6136 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6148 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6160 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6172 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6184 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6196 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6208 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.6220 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6240 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.6253 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6265 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6277 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6288 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6301 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6312 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6324 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6336 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6348 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-filter-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6360 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-subjects-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6381 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6393 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6405 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6417 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6429 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6440 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6452 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6464 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-filter-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6476 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-subjects-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6487 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6508 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6520 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6532 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6567 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6579 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6591 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6610 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6623 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6635 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6647 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6659 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6671 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6682 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6694 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6706 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6717 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6729 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6741 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6753 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.6764 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.6776 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6788 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.6801 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.6813 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: NULL -&gt; c(0, 0.5)
[INFO] 2024-05-22 17:30:42.6825 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: NULL -&gt; png
[INFO] 2024-05-22 17:30:42.6836 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.6848 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: NULL -&gt; plot_20240522_173040
[INFO] 2024-05-22 17:30:42.6860 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: NULL -&gt; 732
[INFO] 2024-05-22 17:30:42.6871 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: NULL -&gt; 1252.8
[INFO] 2024-05-22 17:30:42.6883 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on experiment-name: NULL -&gt; hd1
[INFO] 2024-05-22 17:30:42.6905 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.6917 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6929 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6941 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6953 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6964 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6976 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.6988 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7000 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7012 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7024 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7036 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7047 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.7059 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.7071 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.7083 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.7094 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: c(0, 0.5) -&gt; NULL
[INFO] 2024-05-22 17:30:42.7106 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: png -&gt; NULL
[INFO] 2024-05-22 17:30:42.7118 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.7130 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: plot_20240522_173040 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7142 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: 732 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7154 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: 1252.8 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7165 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on experiment-name: hd1 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7177 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: NULL -&gt; hd1
[INFO] 2024-05-22 17:30:42.7198 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: hd1 -&gt; 
[INFO] 2024-05-22 17:30:42.7218 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.7230 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.7242 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.7254 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.7265 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.7277 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.7289 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.7308 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7320 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7332 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7344 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.7356 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.7367 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.7379 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.7391 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.7411 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.7423 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.7435 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.7997 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.8011 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8031 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8043 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.8055 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8067 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8080 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8092 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8104 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8116 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8128 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8148 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.8160 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8172 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8188 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8209 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8224 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8236 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8250 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8263 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.8275 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8297 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.8309 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8321 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8334 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8346 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8357 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8369 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8381 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8393 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8405 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-filter-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8419 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-subjects-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8440 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8452 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8465 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8476 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8488 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8500 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8512 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8525 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-filter-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8536 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-subjects-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8548 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8569 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8583 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8595 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8628 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8641 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8654 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8673 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8685 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8697 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8709 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8721 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8733 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8745 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8758 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8770 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8782 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8794 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8805 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8817 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.8829 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: NULL -&gt; counts
[INFO] 2024-05-22 17:30:42.8841 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8852 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.8865 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:42.8877 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: NULL -&gt; c(0, 0.5)
[INFO] 2024-05-22 17:30:42.8889 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: NULL -&gt; png
[INFO] 2024-05-22 17:30:42.8901 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.8914 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: NULL -&gt; plot_20240522_173040
[INFO] 2024-05-22 17:30:42.8926 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: NULL -&gt; 732
[INFO] 2024-05-22 17:30:42.8938 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: NULL -&gt; 1252.8
[INFO] 2024-05-22 17:30:42.8950 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on experiment-name: NULL -&gt; hd1
[INFO] 2024-05-22 17:30:42.8971 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.8984 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.8995 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9007 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9019 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9031 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9043 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9054 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9066 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9078 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9091 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9102 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9114 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: counts -&gt; NULL
[INFO] 2024-05-22 17:30:42.9126 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.9138 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.9149 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:42.9161 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: c(0, 0.5) -&gt; NULL
[INFO] 2024-05-22 17:30:42.9173 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: png -&gt; NULL
[INFO] 2024-05-22 17:30:42.9184 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.9196 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: plot_20240522_173040 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9207 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: 732 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9219 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: 1252.8 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9231 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on experiment-name: hd1 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9242 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: NULL -&gt; hd1
[INFO] 2024-05-22 17:30:42.9266 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: hd1 -&gt; counts
[INFO] 2024-05-22 17:30:42.9287 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: counts -&gt; NULL
[INFO] 2024-05-22 17:30:42.9300 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.9312 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.9323 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.9335 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.9347 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:42.9358 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:42.9378 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9389 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9401 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9414 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.9427 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:42.9439 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:42.9451 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.9464 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:42.9484 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:42.9497 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:42.9509 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:56.0226 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:56.1139 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1188 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_overview: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1208 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:56.1228 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1248 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1267 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1287 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1306 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1326 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1346 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1379 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:56.1398 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on available_filters-show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1417 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_active: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1436 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on remove_all_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1456 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1475 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1495 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-remove_filters: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1514 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-collapse: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1534 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:56.1553 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1584 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show_dropmenu: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:56.1602 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on show: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1621 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1641 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1661 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1679 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1699 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1719 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1740 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.1759 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-filter-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1776 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-subjects-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1807 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1824 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd1-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1841 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1859 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd2-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1875 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1891 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-hd3-col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1903 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on minimise_filter_add_vars: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.1915 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on ADTTE-filter-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1926 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on MAE-subjects-var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1938 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1959 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1971 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.1984 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2015 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on row_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2029 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on col_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2041 pid:3767 token:[f2d44648] teal.slice Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2061 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on var_to_add: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2073 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2084 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2096 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2107 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2119 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2130 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2142 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2154 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2165 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2177 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2189 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2200 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2212 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: NULL -&gt; counts
[INFO] 2024-05-22 17:30:56.2224 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2236 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:56.2247 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd: NULL -&gt; 
[INFO] 2024-05-22 17:30:56.2259 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: NULL -&gt; c(0, 0.5)
[INFO] 2024-05-22 17:30:56.2270 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: NULL -&gt; png
[INFO] 2024-05-22 17:30:56.2282 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:56.2293 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: NULL -&gt; plot_20240522_173040
[INFO] 2024-05-22 17:30:56.2305 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2316 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: NULL -&gt; 0.87
[INFO] 2024-05-22 17:30:56.2328 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on experiment-name: NULL -&gt; hd1
[INFO] 2024-05-22 17:30:56.2349 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2361 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-add_report_card_simple-add_report_card_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2373 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-download_button_simple-download_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2385 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-archive_load_simple-reporter_load: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2397 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on simple_reporter-reset_button_simple-reset_reporter: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2409 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2421 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2432 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2444 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on strata-levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2455 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-downl: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2468 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expand: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2479 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-expbut: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2491 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on assay-name: counts -&gt; NULL
[INFO] 2024-05-22 17:30:56.2503 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2514 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:56.2526 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on adtte-paramcd:  -&gt; NULL
[INFO] 2024-05-22 17:30:56.2538 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on percentiles: c(0, 0.5) -&gt; NULL
[INFO] 2024-05-22 17:30:56.2550 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_format: png -&gt; NULL
[INFO] 2024-05-22 17:30:56.2562 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes-lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:56.2576 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-downbutton-file_name: plot_20240522_173040 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2591 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-flex_width: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2606 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on plot-plot_modal_width: 0.87 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2618 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on experiment-name: hd1 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2630 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: NULL -&gt; hd1
[INFO] 2024-05-22 17:30:56.2651 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: hd1 -&gt; counts
[INFO] 2024-05-22 17:30:56.2671 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on name: counts -&gt; NULL
[INFO] 2024-05-22 17:30:56.2683 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2695 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2707 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2718 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2730 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: NULL -&gt; Mean
[INFO] 2024-05-22 17:30:56.2741 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: NULL -&gt; FALSE
[INFO] 2024-05-22 17:30:56.2760 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_none_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2772 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on select_all_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2784 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on text_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2796 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on genes: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2807 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on fun_name: Mean -&gt; NULL
[INFO] 2024-05-22 17:30:56.2819 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on lock_button: FALSE -&gt; NULL
[INFO] 2024-05-22 17:30:56.2831 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2843 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: NULL -&gt; 0
[INFO] 2024-05-22 17:30:56.2863 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on sample_var: NULL -&gt; NULL
[INFO] 2024-05-22 17:30:56.2875 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on levels_button: 0 -&gt; NULL
[INFO] 2024-05-22 17:30:56.2887 pid:3767 token:[f2d44648] teal.modules.hermes Shiny input change detected on paramcd: NULL -&gt; </code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="rnag09_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] random.cdisc.data_0.3.15.9003  teal.modules.hermes_0.1.6.9015
 [3] teal_0.15.2.9038               teal.slice_0.5.1.9003         
 [5] teal.data_0.6.0.9005           teal.code_0.5.0.9005          
 [7] shiny_1.8.1.1                  hermes_1.7.1.9002             
 [9] SummarizedExperiment_1.34.0    Biobase_2.64.0                
[11] GenomicRanges_1.56.0           GenomeInfoDb_1.40.0           
[13] IRanges_2.38.0                 S4Vectors_0.42.0              
[15] BiocGenerics_0.50.0            MatrixGenerics_1.16.0         
[17] matrixStats_1.3.0              ggfortify_0.4.17              
[19] ggplot2_3.5.1                  dplyr_1.1.4                   
[21] tern_0.9.4.9007                rtables_0.6.7.9002            
[23] magrittr_2.0.3                 formatters_0.5.6.9004         

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3          jsonlite_1.8.8             
  [3] shape_1.4.6.1               MultiAssayExperiment_1.30.1
  [5] farver_2.1.2                rmarkdown_2.27             
  [7] GlobalOptions_0.1.2         zlibbioc_1.50.0            
  [9] vctrs_0.6.5                 memoise_2.0.1              
 [11] webshot_0.5.5               BiocBaseUtils_1.7.0        
 [13] htmltools_0.5.8.1           S4Arrays_1.4.0             
 [15] forcats_1.0.0               progress_1.2.3             
 [17] curl_5.2.1                  broom_1.0.6                
 [19] SparseArray_1.4.3           sass_0.4.9                 
 [21] bslib_0.7.0                 fontawesome_0.5.2          
 [23] htmlwidgets_1.6.4           httr2_1.0.1                
 [25] cachem_1.1.0                teal.widgets_0.4.2.9011    
 [27] mime_0.12                   lifecycle_1.0.4            
 [29] iterators_1.0.14            pkgconfig_2.0.3            
 [31] webshot2_0.1.1              Matrix_1.7-0               
 [33] R6_2.5.1                    fastmap_1.2.0              
 [35] GenomeInfoDbData_1.2.12     rbibutils_2.2.16           
 [37] clue_0.3-65                 digest_0.6.35              
 [39] colorspace_2.1-0            shinycssloaders_1.0.0      
 [41] ps_1.7.6                    AnnotationDbi_1.66.0       
 [43] RSQLite_2.3.6               filelock_1.0.3             
 [45] labeling_0.4.3              fansi_1.0.6                
 [47] httr_1.4.7                  abind_1.4-5                
 [49] compiler_4.4.0              bit64_4.0.5                
 [51] withr_3.0.0                 doParallel_1.0.17          
 [53] backports_1.4.1             DBI_1.2.2                  
 [55] logger_0.3.0                biomaRt_2.60.0             
 [57] rappdirs_0.3.3              DelayedArray_0.30.1        
 [59] rjson_0.2.21                chromote_0.2.0             
 [61] tools_4.4.0                 httpuv_1.6.15              
 [63] glue_1.7.0                  callr_3.7.6                
 [65] promises_1.3.0              grid_4.4.0                 
 [67] checkmate_2.3.1             cluster_2.1.6              
 [69] generics_0.1.3              gtable_0.3.5               
 [71] websocket_1.4.1             tidyr_1.3.1                
 [73] hms_1.1.3                   xml2_1.3.6                 
 [75] utf8_1.2.4                  XVector_0.44.0             
 [77] ggrepel_0.9.5               foreach_1.5.2              
 [79] pillar_1.9.0                stringr_1.5.1              
 [81] later_1.3.2                 circlize_0.4.16            
 [83] splines_4.4.0               BiocFileCache_2.12.0       
 [85] lattice_0.22-6              survival_3.6-4             
 [87] bit_4.0.5                   tidyselect_1.2.1           
 [89] ComplexHeatmap_2.20.0       Biostrings_2.72.0          
 [91] knitr_1.46                  gridExtra_2.3              
 [93] teal.logger_0.2.0.9002      xfun_0.44                  
 [95] stringi_1.8.4               UCSC.utils_1.0.0           
 [97] yaml_2.3.8                  shinyWidgets_0.8.6         
 [99] evaluate_0.23               codetools_0.2-20           
[101] tibble_3.2.1                cli_3.6.2                  
[103] xtable_1.8-4                Rdpack_2.6                 
[105] jquerylib_0.1.4             processx_3.8.4             
[107] munsell_0.5.1               teal.reporter_0.3.1.9006   
[109] Rcpp_1.0.12                 dbplyr_2.5.0               
[111] png_0.1-8                   parallel_4.4.0             
[113] assertthat_0.2.1            blob_1.2.4                 
[115] prettyunits_1.2.0           scales_1.3.0               
[117] purrr_1.0.2                 crayon_1.5.2               
[119] GetoptLong_1.0.5            rlang_1.1.3                
[121] formatR_1.14                cowplot_1.1.3              
[123] KEGGREST_1.44.0             shinyjs_2.1.0              </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- -->


<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div>Copyright 2023, Hoffmann-La Roche Ltd.</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/insightsengineering\.github\.io\/biomarker-catalog");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><script src="https://giscus.app/client.js" data-repo="insightsengineering/biomarker-catalog" data-repo-id="R_kgDOJMeENQ" data-category="General" data-category-id="DIC_kwDOJMeENc4CVB9s" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" data-loading="lazy" async="">
</script><input type="hidden" id="giscus-base-theme" value="light"><input type="hidden" id="giscus-alt-theme" value="light"><nav class="page-navigation column-page-right"><div class="nav-page nav-page-previous">
      <a href="../graphs/rnag08.html" class="pagination-link" aria-label="RNAG8">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">RNAG8</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../graphs/rnag10.html" class="pagination-link" aria-label="RNAG10">
        <span class="nav-page-text">RNAG10</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb11" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">---</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="an">title:</span><span class="co"> RNAG9</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="an">subtitle:</span><span class="co"> RNAseq Kaplan-Meier Graph</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="an">categories:</span><span class="co"> [RNAG]</span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="co">---</span></span>
<span id="cb11-6"><a href="#cb11-6"></a></span>
<span id="cb11-7"><a href="#cb11-7"></a>------------------------------------------------------------------------</span>
<span id="cb11-8"><a href="#cb11-8"></a></span>
<span id="cb11-9"><a href="#cb11-9"></a>{{&lt; include ../misc/hooks.qmd &gt;}}</span>
<span id="cb11-10"><a href="#cb11-10"></a></span>
<span id="cb11-11"><a href="#cb11-11"></a>::: panel-tabset</span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="fu">## Setup: Import &amp; Data Manipulation</span></span>
<span id="cb11-13"><a href="#cb11-13"></a></span>
<span id="cb11-14"><a href="#cb11-14"></a>This page can be used as a template of how to produce Kaplan-Meier graphs for RNA-seq gene expression analysis using available <span class="in">`tern`</span> and <span class="in">`hermes`</span> functions, and to create an interactive Kaplan-Meier graph for RNA-seq gene expression analysis using <span class="in">`teal.modules.hermes`</span>.</span>
<span id="cb11-15"><a href="#cb11-15"></a></span>
<span id="cb11-16"><a href="#cb11-16"></a>The code below needs both RNA-seq data (in <span class="in">`HermesData`</span> format) and time-to-event data (in <span class="in">`ADTTE`</span> format) as input.</span>
<span id="cb11-17"><a href="#cb11-17"></a></span>
<span id="cb11-18"><a href="#cb11-18"></a>We first prepare the time-to-event data.</span>
<span id="cb11-19"><a href="#cb11-19"></a>We define an event indicator variable, transform the time to months and filter down to the overall survival subset.</span>
<span id="cb11-20"><a href="#cb11-20"></a></span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="in">```{r, message = FALSE}</span></span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="in">library(tern)</span></span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="in">library(dplyr)</span></span>
<span id="cb11-24"><a href="#cb11-24"></a></span>
<span id="cb11-25"><a href="#cb11-25"></a><span class="in">adtte_f &lt;- random.cdisc.data::cadtte %&gt;%</span></span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="in">  dplyr::mutate(</span></span>
<span id="cb11-27"><a href="#cb11-27"></a><span class="in">    is_event = .data$CNSR == 0,</span></span>
<span id="cb11-28"><a href="#cb11-28"></a><span class="in">    AVAL = day2month(.data$AVAL),</span></span>
<span id="cb11-29"><a href="#cb11-29"></a><span class="in">    AVALU = "Months"</span></span>
<span id="cb11-30"><a href="#cb11-30"></a><span class="in">  ) %&gt;%</span></span>
<span id="cb11-31"><a href="#cb11-31"></a><span class="in">  dplyr::filter(.data$PARAMCD == "OS")</span></span>
<span id="cb11-32"><a href="#cb11-32"></a><span class="in">```</span></span>
<span id="cb11-33"><a href="#cb11-33"></a></span>
<span id="cb11-34"><a href="#cb11-34"></a>Then we prepare the RNA-seq data.</span>
<span id="cb11-35"><a href="#cb11-35"></a>See <span class="co">[</span><span class="ot">RNAG1</span><span class="co">](../graphs/rnag01.qmd)</span> for basic details on how to import, filter and normalize <span class="in">`HermesData`</span>.</span>
<span id="cb11-36"><a href="#cb11-36"></a>We use <span class="in">`col_data_with_genes()`</span> to extract the sample variables (<span class="in">`colData`</span>) from the object, together with a single specified gene or a specified gene signature.</span>
<span id="cb11-37"><a href="#cb11-37"></a>See <span class="in">`?hermes::gene_spec`</span> for details on how to do this.</span>
<span id="cb11-38"><a href="#cb11-38"></a>Then we use <span class="in">`inner_join_cdisc()`</span> to join this genetic data with the <span class="in">`ADTTE`</span> data from above.</span>
<span id="cb11-39"><a href="#cb11-39"></a>See the help page for more details, in particular how the join keys can be customized if needed - here we just join based on <span class="in">`USUBJID`</span> by default.</span>
<span id="cb11-40"><a href="#cb11-40"></a></span>
<span id="cb11-41"><a href="#cb11-41"></a><span class="in">```{r,  message = FALSE}</span></span>
<span id="cb11-42"><a href="#cb11-42"></a><span class="in">library(hermes)</span></span>
<span id="cb11-43"><a href="#cb11-43"></a></span>
<span id="cb11-44"><a href="#cb11-44"></a><span class="in">gene_data &lt;- col_data_with_genes(</span></span>
<span id="cb11-45"><a href="#cb11-45"></a><span class="in">  object = hermes_data,</span></span>
<span id="cb11-46"><a href="#cb11-46"></a><span class="in">  assay_name = "counts",</span></span>
<span id="cb11-47"><a href="#cb11-47"></a><span class="in">  genes = gene_spec("GeneID:1820")</span></span>
<span id="cb11-48"><a href="#cb11-48"></a><span class="in">)</span></span>
<span id="cb11-49"><a href="#cb11-49"></a><span class="in">joined_data &lt;- inner_join_cdisc(</span></span>
<span id="cb11-50"><a href="#cb11-50"></a><span class="in">  gene_data = gene_data,</span></span>
<span id="cb11-51"><a href="#cb11-51"></a><span class="in">  cdisc_data = adtte_f</span></span>
<span id="cb11-52"><a href="#cb11-52"></a><span class="in">)</span></span>
<span id="cb11-53"><a href="#cb11-53"></a><span class="in">```</span></span>
<span id="cb11-54"><a href="#cb11-54"></a></span>
<span id="cb11-55"><a href="#cb11-55"></a>We can then cut the resulting gene column (we figure out the column name and save it in <span class="in">`arm_name`</span> below) in the <span class="in">`joined_data`</span> into quantile bins (in this example we want three equally sized groups).</span>
<span id="cb11-56"><a href="#cb11-56"></a></span>
<span id="cb11-59"><a href="#cb11-59"></a><span class="in">```{r}</span></span>
<span id="cb11-60"><a href="#cb11-60"></a>arm_name <span class="ot">&lt;-</span> <span class="fu">attr</span>(gene_data, <span class="st">"gene_cols"</span>)</span>
<span id="cb11-61"><a href="#cb11-61"></a>binned_data <span class="ot">&lt;-</span> joined_data <span class="sc">%&gt;%</span></span>
<span id="cb11-62"><a href="#cb11-62"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb11-63"><a href="#cb11-63"></a>    <span class="at">gene_factor =</span> <span class="fu">cut_quantile_bins</span>(</span>
<span id="cb11-64"><a href="#cb11-64"></a>      joined_data[[arm_name]],</span>
<span id="cb11-65"><a href="#cb11-65"></a>      <span class="at">probs =</span> <span class="fu">c</span>(<span class="dv">1</span> <span class="sc">/</span> <span class="dv">3</span>, <span class="dv">2</span> <span class="sc">/</span> <span class="dv">3</span>)</span>
<span id="cb11-66"><a href="#cb11-66"></a>    )</span>
<span id="cb11-67"><a href="#cb11-67"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb11-68"><a href="#cb11-68"></a>  <span class="fu">df_explicit_na</span>()</span>
<span id="cb11-69"><a href="#cb11-69"></a><span class="in">```</span></span>
<span id="cb11-70"><a href="#cb11-70"></a></span>
<span id="cb11-71"><a href="#cb11-71"></a><span class="fu">## Kaplan-Meier Graph</span></span>
<span id="cb11-72"><a href="#cb11-72"></a></span>
<span id="cb11-73"><a href="#cb11-73"></a>It is now simple to create the Kaplan-Meier graph by providing the data set created above with the variable specification.</span>
<span id="cb11-74"><a href="#cb11-74"></a>Note that we specify the above created <span class="in">`gene_factor`</span> as arm variable here.</span>
<span id="cb11-75"><a href="#cb11-75"></a></span>
<span id="cb11-78"><a href="#cb11-78"></a><span class="in">```{r}</span></span>
<span id="cb11-79"><a href="#cb11-79"></a>variables <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb11-80"><a href="#cb11-80"></a>  <span class="at">tte =</span> <span class="st">"AVAL"</span>,</span>
<span id="cb11-81"><a href="#cb11-81"></a>  <span class="at">is_event =</span> <span class="st">"is_event"</span>,</span>
<span id="cb11-82"><a href="#cb11-82"></a>  <span class="at">arm =</span> <span class="st">"gene_factor"</span></span>
<span id="cb11-83"><a href="#cb11-83"></a>)</span>
<span id="cb11-84"><a href="#cb11-84"></a><span class="fu">g_km</span>(</span>
<span id="cb11-85"><a href="#cb11-85"></a>  binned_data,</span>
<span id="cb11-86"><a href="#cb11-86"></a>  <span class="at">variables =</span> variables,</span>
<span id="cb11-87"><a href="#cb11-87"></a>  <span class="at">annot_coxph =</span> <span class="cn">TRUE</span></span>
<span id="cb11-88"><a href="#cb11-88"></a>)</span>
<span id="cb11-89"><a href="#cb11-89"></a><span class="in">```</span></span>
<span id="cb11-90"><a href="#cb11-90"></a></span>
<span id="cb11-91"><a href="#cb11-91"></a>See <span class="co">[</span><span class="ot">KG1</span><span class="co">](../graphs/KG1/kg01.qmd)</span> to <span class="co">[</span><span class="ot">KG5</span><span class="co">](../graphs/KG5/kg05.qmd)</span> for additional customization options for the Kaplan-Meier graphs or the help page <span class="in">`?g_km()`</span>.</span>
<span id="cb11-92"><a href="#cb11-92"></a></span>
<span id="cb11-93"><a href="#cb11-93"></a><span class="fu">## Teal Module for Kaplan-Meier Graph</span></span>
<span id="cb11-94"><a href="#cb11-94"></a></span>
<span id="cb11-95"><a href="#cb11-95"></a>We start by importing a <span class="in">`MultiAssayExperiment`</span> and sample <span class="in">`ADTTE`</span> data; here we use the example <span class="in">`multi_assay_experiment`</span> available in <span class="in">`hermes`</span> and example <span class="in">`ADTTE`</span> data from <span class="in">`random.cdisc.data`</span>.</span>
<span id="cb11-96"><a href="#cb11-96"></a>We can then use the provided teal module <span class="in">`tm_g_km`</span> to include the corresponding interactive Kaplan-Meier analysis in our teal app.</span>
<span id="cb11-97"><a href="#cb11-97"></a>Note that by default the <span class="in">`counts`</span> assay is excluded via the <span class="in">`exclude_assays`</span> argument, but we can include it by just saying that we don't want to exclude any assays.</span>
<span id="cb11-98"><a href="#cb11-98"></a>In case that we have different non-standard column names in our ADTTE data set we could also specify them via the <span class="in">`adtte_vars`</span> argument, see the documentation <span class="in">`?tm_g_km`</span> for more details.</span>
<span id="cb11-99"><a href="#cb11-99"></a></span>
<span id="cb11-100"><a href="#cb11-100"></a><span class="in">```{r,  message = FALSE, cache = FALSE, opts.label=c('app')}</span></span>
<span id="cb11-101"><a href="#cb11-101"></a><span class="in">library(teal.modules.hermes)</span></span>
<span id="cb11-102"><a href="#cb11-102"></a></span>
<span id="cb11-103"><a href="#cb11-103"></a><span class="in">data &lt;- teal_data()</span></span>
<span id="cb11-104"><a href="#cb11-104"></a><span class="in">data &lt;- within(data, {</span></span>
<span id="cb11-105"><a href="#cb11-105"></a><span class="in">  library(random.cdisc.data)</span></span>
<span id="cb11-106"><a href="#cb11-106"></a><span class="in">  library(dplyr)</span></span>
<span id="cb11-107"><a href="#cb11-107"></a><span class="in">  library(hermes)</span></span>
<span id="cb11-108"><a href="#cb11-108"></a><span class="in">  MAE &lt;- multi_assay_experiment</span></span>
<span id="cb11-109"><a href="#cb11-109"></a><span class="in">  ADTTE &lt;- random.cdisc.data::cadtte %&gt;%</span></span>
<span id="cb11-110"><a href="#cb11-110"></a><span class="in">    mutate(is_event = .data$CNSR == 0)</span></span>
<span id="cb11-111"><a href="#cb11-111"></a><span class="in">})</span></span>
<span id="cb11-112"><a href="#cb11-112"></a><span class="in">datanames(data) &lt;- c("MAE", "ADTTE")</span></span>
<span id="cb11-113"><a href="#cb11-113"></a><span class="in">join_keys(data)["ADTTE", "ADTTE"] &lt;- c("STUDYID", "USUBJID", "PARAMCD")</span></span>
<span id="cb11-114"><a href="#cb11-114"></a></span>
<span id="cb11-115"><a href="#cb11-115"></a><span class="in">app &lt;- init(</span></span>
<span id="cb11-116"><a href="#cb11-116"></a><span class="in">  data = data,</span></span>
<span id="cb11-117"><a href="#cb11-117"></a><span class="in">  modules = modules(</span></span>
<span id="cb11-118"><a href="#cb11-118"></a><span class="in">    tm_g_km(</span></span>
<span id="cb11-119"><a href="#cb11-119"></a><span class="in">      label = "Kaplan-Meier Graph",</span></span>
<span id="cb11-120"><a href="#cb11-120"></a><span class="in">      adtte_name = "ADTTE",</span></span>
<span id="cb11-121"><a href="#cb11-121"></a><span class="in">      mae_name = "MAE",</span></span>
<span id="cb11-122"><a href="#cb11-122"></a><span class="in">      exclude_assays = character()</span></span>
<span id="cb11-123"><a href="#cb11-123"></a><span class="in">    )</span></span>
<span id="cb11-124"><a href="#cb11-124"></a><span class="in">  )</span></span>
<span id="cb11-125"><a href="#cb11-125"></a><span class="in">)</span></span>
<span id="cb11-126"><a href="#cb11-126"></a><span class="in">shinyApp(app$ui, app$server)</span></span>
<span id="cb11-127"><a href="#cb11-127"></a><span class="in">```</span></span>
<span id="cb11-128"><a href="#cb11-128"></a></span>
<span id="cb11-129"><a href="#cb11-129"></a>{{&lt; include ../misc/session_info.qmd &gt;}}</span>
<span id="cb11-130"><a href="#cb11-130"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Made with ❤️ by the Statistical Engineering Team <a href="https://github.com/insightsengineering"><img src="https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/SVG/StatisticalEngineering.svg" class="img-fluid" alt="StatisticalEngineering" width="15"></a></p>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="../misc/license.html">
<p>License</p>
</a>
  </li>  
</ul>
<div class="toc-actions"><ul><li><a href="https://github.com/insightsengineering/biomarker-catalog/edit/main/graphs/rnag09.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/insightsengineering/biomarker-catalog/issues/new/choose" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/insightsengineering/biomarker-catalog">
      <i class="bi bi-github" role="img" aria-label="BAC GitHub">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://insightsengineering.github.io/biomarker-catalog/index.xml">
      <i class="bi bi-rss" role="img" aria-label="BAC RSS">
</i> 
    </a>
  </li>  
</ul>
</div>
  </div>
</footer>


<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>