<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Biomarker Analysis Catalog - RNAG5</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../graphs/rnag06.html" rel="next">
<link href="../graphs/rnag04.html" rel="prev">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": true,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script><script type="text/plain" cookie-consent="tracking">

(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-125641273-1', 'auto');

ga('send', {
  hitType: 'pageview',
  'anonymizeIp': true,
});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<link rel="stylesheet" href="../assets/css/style.scss">
<meta property="og:title" content="Biomarker Analysis Catalog - RNAG5">
<meta property="og:description" content="RNAseq Differential Gene Expression Analysis">
<meta property="og:image" content="https://insightsengineering.github.io/biomarker-catalog/graphs/rnag05_files/figure-html/unnamed-chunk-5-1.png">
<meta property="og:site-name" content="Biomarker Analysis Catalog">
<meta property="og:image:height" content="960">
<meta property="og:image:width" content="2880">
</head>
<body class="nav-sidebar docked nav-fixed">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Biomarker Analysis Catalog</span>
    </a>
  </div>
          <ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://app.slack.com/client/T028PB489D3/" rel="" target=""><i class="bi bi-slack" role="img" aria-label="Slack">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/insightsengineering/biomarker-catalog" rel="" target=""><i class="bi bi-github" role="img" aria-label="Repository">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
            <div id="quarto-search" class="" title="true"></div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../graphs/index.html">Graphs</a></li><li class="breadcrumb-item"><a href="../graphs/rnag05.html">RNAG5</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../assets/images/logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../tables/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tables</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/cmt01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CMT1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/cmt02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CMT2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/cmt03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CMT3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/dt01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/dt02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DT2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tables/tet01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">TET1</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../graphs/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Graphs</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/ag01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">AG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/dg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/dg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/dg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/dg04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DG4</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/kg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/kg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/kg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/kg04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG4</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/kg05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KG5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rfg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rfg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rfg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RFG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG4</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag05.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">RNAG5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag06.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG6</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag07.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG7</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag08.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG8</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag09.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG9</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/rnag10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAG10</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg03.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG4</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/sfg06.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SFG6</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/spg01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SPG1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../graphs/spg02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SPG2</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../bac_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Index</span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content column-page-right" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">RNAG5</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
<p class="subtitle lead">RNAseq Differential Gene Expression Analysis</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><hr>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">Setup: Import and Filter</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Differential Gene Expression Analysis via Limma-Voom Method</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">Differential Gene Expression Analysis via DESeq2 Method</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">Additional Arguments</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-5" role="tab" aria-controls="tabset-1-5" aria-selected="false">Plot Differential Gene Expression in Volcano Graph</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-6" role="tab" aria-controls="tabset-1-6" aria-selected="false">Teal Module for Sample Differential Gene Expression</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-7-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-7" role="tab" aria-controls="tabset-1-7" aria-selected="false">Session Info</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<p>This page can be used as a template of how to use the available <code>hermes</code> functions to take differential gene expression analysis between samples in <code>HermesData</code>, visualize them in a volcano graph, and create an interactive volcano graph using <code>teal.module.hermes</code>.</p>
<p>The function used to calculate differential gene expression uses <code>HermesData</code> as input. See <a href="../graphs/rnag01.html">RNAG1</a> for details. In particular, the differential gene expression analysis does not require to normalize <code>HermesData</code>.</p>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-1_363685ed43f7bf28cc6cc3e68860e69a">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(hermes)</span>
<span id="cb1-2"><a href="#cb1-2"></a></span>
<span id="cb1-3"><a href="#cb1-3"></a>object <span class="ot">&lt;-</span> hermes_data <span class="sc">%&gt;%</span></span>
<span id="cb1-4"><a href="#cb1-4"></a>  <span class="fu">add_quality_flags</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-5"><a href="#cb1-5"></a>  <span class="fu">filter</span>()</span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">colData</span>(object) <span class="ot">&lt;-</span> <span class="fu">df_cols_to_factor</span>(<span class="fu">colData</span>(object))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<p>We can calculate a data frame in <code>HermesDataDiffExpr</code> class to include the gene differential expression between groups in <code>HermesData</code> using <code>diff_expression()</code> function. By default, <code>diff_expression()</code> function uses the <code>limma-voom</code> method to calculate the gene differential expression, unless specified otherwise as argument. The <code>group</code> argument is to set the groups for comparison. The <code>group</code> factors should be a 2-level factor in <code>colData(HermesData)</code>.</p>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-2_929c53622d49b4988024e283ef0fc528">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>result1 <span class="ot">&lt;-</span> <span class="fu">diff_expression</span>(object, <span class="at">group =</span> <span class="st">"SEX"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<p>We can also use the <code>DESeq2</code> Method for gene differential expression.</p>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-3_d6a842dca6b45b6281842b6dfc3a3461">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>result2 <span class="ot">&lt;-</span> <span class="fu">diff_expression</span>(object, <span class="at">group =</span> <span class="st">"SEX"</span>, <span class="at">method =</span> <span class="st">"deseq2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<p>It is also possible to pass additional arguments to the function behind the methods in the <code>diff_expression()</code>, ex. if we wish to change the type of size factor estimation when using the <code>DESeq2</code> method. See <code>?DESeq2::DESeq()</code> or <code>?limma::eBayes()</code> for details about the additional parameters available.</p>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-4_046d171eb7625575fe5cc0ab1b9b5cc1">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>result3 <span class="ot">&lt;-</span> <span class="fu">diff_expression</span>(object, <span class="at">group =</span> <span class="st">"SEX"</span>, <span class="at">method =</span> <span class="st">"deseq2"</span>, <span class="at">sfType =</span> <span class="st">"ratio"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div id="tabset-1-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-5-tab">
<p>We can then plot a volcano graph for differential gene expression between groups in <code>HermesData</code> using the <code>autoplot()</code> function. The threshold values for adjusted p-values and log2 folder change can be specified.</p>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-5_1b13b278ba9a88de87148d20cea9a463">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">autoplot</span>(result1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="rnag05_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-6_9debe9730c19305efc7668eaeb5ad9f0">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">autoplot</span>(result2, <span class="at">adj_p_val_thresh =</span> <span class="fl">0.025</span>, <span class="at">log2_fc_thresh =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="rnag05_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div class="cell" data-hash="rnag05_cache/html/unnamed-chunk-7_76696d2db665dcb9af69833a3909349c">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="fu">autoplot</span>(result3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 47 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="rnag05_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
</div>
<div id="tabset-1-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-6-tab">
<p>We start by importing a <code>MultiAssayExperiment</code>; here we use the example <code>multi_assay_experiment</code> available in <code>hermes</code>. It is wrapped as a <code>teal::dataset</code>. We can then use the provided teal module <code>tm_g_volcanoplot</code> to include a volcano graph module in our teal app.</p>
<div class="cell" data-screenshot.opts="{&quot;delay&quot;:24}">
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">library</span>(teal.modules.hermes)</span>
<span id="cb10-2"><a href="#cb10-2"></a>mae <span class="ot">&lt;-</span> hermes<span class="sc">::</span>multi_assay_experiment</span>
<span id="cb10-3"><a href="#cb10-3"></a>mae_data <span class="ot">&lt;-</span> <span class="fu">dataset</span>(<span class="st">"MAE"</span>, mae)</span>
<span id="cb10-4"><a href="#cb10-4"></a>data <span class="ot">&lt;-</span> <span class="fu">teal_data</span>(mae_data)</span>
<span id="cb10-5"><a href="#cb10-5"></a></span>
<span id="cb10-6"><a href="#cb10-6"></a>app <span class="ot">&lt;-</span> <span class="fu">init</span>(</span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="at">data =</span> data,</span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="at">modules =</span> <span class="fu">modules</span>(</span>
<span id="cb10-9"><a href="#cb10-9"></a>    <span class="fu">tm_g_volcanoplot</span>(</span>
<span id="cb10-10"><a href="#cb10-10"></a>      <span class="at">label =</span> <span class="st">"volcanoplot"</span>,</span>
<span id="cb10-11"><a href="#cb10-11"></a>      <span class="at">mae_name =</span> <span class="st">"MAE"</span></span>
<span id="cb10-12"><a href="#cb10-12"></a>    )</span>
<span id="cb10-13"><a href="#cb10-13"></a>  )</span>
<span id="cb10-14"><a href="#cb10-14"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[INFO] 2023-05-27 17:42:00.6799 pid:3876 token:[] teal.modules.hermes Initializing tm_g_volcanoplot</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">shinyApp</span>(app<span class="sc">$</span>ui, app<span class="sc">$</span>server)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: 'experiments' dropped; see 'metadata'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="rnag05_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
</div>
<div id="tabset-1-7" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-7-tab">
<div class="cell" data-hash="rnag05_cache/html/sessioninfo_fad61d7fb1f4c7dd1c0886aef65dfb45">
<details open=""><summary>Code</summary><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] teal.modules.hermes_0.1.4.9010 teal_0.13.0                   
 [3] teal.transform_0.3.0           magrittr_2.0.3                
 [5] teal.data_0.2.0                shiny_1.7.4                   
 [7] hermes_1.0.1.9011              SummarizedExperiment_1.28.0   
 [9] Biobase_2.58.0                 GenomicRanges_1.50.2          
[11] GenomeInfoDb_1.34.9            IRanges_2.32.0                
[13] S4Vectors_0.36.2               BiocGenerics_0.44.0           
[15] MatrixGenerics_1.10.0          matrixStats_0.63.0            
[17] ggfortify_0.4.16               ggplot2_3.4.2                 

loaded via a namespace (and not attached):
  [1] backports_1.4.1             circlize_0.4.15            
  [3] systemfonts_1.0.4           BiocBaseUtils_1.3.0        
  [5] BiocFileCache_2.6.1         crosstalk_1.2.0            
  [7] BiocParallel_1.32.6         teal.logger_0.1.1.9004     
  [9] digest_0.6.31               foreach_1.5.2              
 [11] htmltools_0.5.5             fansi_1.0.4                
 [13] checkmate_2.2.0             memoise_2.0.1              
 [15] cluster_2.1.4               doParallel_1.0.17          
 [17] limma_3.54.2                ComplexHeatmap_2.14.0      
 [19] Biostrings_2.66.0           annotate_1.76.0            
 [21] prettyunits_1.1.1           colorspace_2.1-0           
 [23] blob_1.2.4                  rappdirs_0.3.3             
 [25] ggrepel_0.9.3               textshaping_0.3.6          
 [27] rbibutils_2.2.13            xfun_0.39                  
 [29] dplyr_1.1.2                 callr_3.7.3                
 [31] crayon_1.5.2                RCurl_1.98-1.12            
 [33] jsonlite_1.8.4              iterators_1.0.14           
 [35] glue_1.6.2                  gtable_0.3.3               
 [37] zlibbioc_1.44.0             XVector_0.38.0             
 [39] webshot_0.5.4               GetoptLong_1.0.5           
 [41] DelayedArray_0.24.0         shape_1.4.6                
 [43] scales_1.2.1                DBI_1.1.3                  
 [45] edgeR_3.40.2                Rcpp_1.0.10                
 [47] xtable_1.8-4                progress_1.2.2             
 [49] clue_0.3-64                 bit_4.0.5                  
 [51] DT_0.28                     htmlwidgets_1.6.2          
 [53] httr_1.4.6                  RColorBrewer_1.1-3         
 [55] ellipsis_0.3.2              teal.slice_0.3.0           
 [57] pkgconfig_2.0.3             XML_3.99-0.14              
 [59] farver_2.1.1                sass_0.4.6                 
 [61] dbplyr_2.3.2                locfit_1.5-9.7             
 [63] utf8_1.2.3                  tidyselect_1.2.0           
 [65] labeling_0.4.2              rlang_1.1.1                
 [67] later_1.3.1                 AnnotationDbi_1.60.2       
 [69] munsell_0.5.0               tools_4.2.2                
 [71] cachem_1.0.8                cli_3.6.1                  
 [73] generics_0.1.3              RSQLite_2.3.1              
 [75] evaluate_0.21               stringr_1.5.0              
 [77] fastmap_1.1.1               ragg_1.2.5                 
 [79] yaml_2.3.7                  processx_3.8.1             
 [81] knitr_1.42                  bit64_4.0.5                
 [83] purrr_1.0.1                 KEGGREST_1.38.0            
 [85] teal.reporter_0.1.1.9014    mime_0.12                  
 [87] xml2_1.3.4                  biomaRt_2.54.1             
 [89] brio_1.1.3                  compiler_4.2.2             
 [91] filelock_1.0.2              curl_5.0.0                 
 [93] png_0.1-8                   testthat_3.1.8             
 [95] tibble_3.2.1                geneplotter_1.76.0         
 [97] bslib_0.4.2                 stringi_1.7.12             
 [99] ps_1.7.5                    forcats_1.0.0              
[101] logger_0.2.2                lattice_0.21-8             
[103] Matrix_1.5-4.1              shinyjs_2.1.0              
[105] vctrs_0.6.2                 pillar_1.9.0               
[107] lifecycle_1.0.3             Rdpack_2.4                 
[109] jquerylib_0.1.4             GlobalOptions_0.1.2        
[111] bitops_1.0-7                httpuv_1.6.11              
[113] R6_2.5.1                    teal.widgets_0.3.0         
[115] promises_1.2.0.1            gridExtra_2.3              
[117] codetools_0.2-19            formatters_0.5.0           
[119] assertthat_0.2.1            fontawesome_0.5.1          
[121] DESeq2_1.38.3               rjson_0.2.21               
[123] shinyWidgets_0.7.6          withr_2.5.0                
[125] GenomeInfoDbData_1.2.9      parallel_4.2.2             
[127] hms_1.1.3                   MultiAssayExperiment_1.24.0
[129] grid_4.2.2                  tidyr_1.3.0                
[131] rmarkdown_2.21             </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- -->


</main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><script src="https://giscus.app/client.js" data-repo="insightsengineering/biomarker-catalog" data-repo-id="R_kgDOJMeENQ" data-category="General" data-category-id="DIC_kwDOJMeENc4CVB9s" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" data-loading="lazy" async="">
</script><nav class="page-navigation column-page-right"><div class="nav-page nav-page-previous">
      <a href="../graphs/rnag04.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">RNAG4</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../graphs/rnag06.html" class="pagination-link">
        <span class="nav-page-text">RNAG6</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb16" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb16-1"><a href="#cb16-1"></a><span class="co">---</span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="an">title:</span><span class="co"> RNAG5</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="an">subtitle:</span><span class="co"> RNAseq Differential Gene Expression Analysis</span></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="co">---</span></span>
<span id="cb16-5"><a href="#cb16-5"></a></span>
<span id="cb16-6"><a href="#cb16-6"></a>------------------------------------------------------------------------</span>
<span id="cb16-7"><a href="#cb16-7"></a></span>
<span id="cb16-8"><a href="#cb16-8"></a>::: panel-tabset</span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="fu">## Setup: Import and Filter</span></span>
<span id="cb16-10"><a href="#cb16-10"></a></span>
<span id="cb16-11"><a href="#cb16-11"></a>This page can be used as a template of how to use the available <span class="in">`hermes`</span> functions to take differential gene expression analysis between samples in <span class="in">`HermesData`</span>, visualize them in a volcano graph, and create an interactive volcano graph using <span class="in">`teal.module.hermes`</span>.</span>
<span id="cb16-12"><a href="#cb16-12"></a></span>
<span id="cb16-13"><a href="#cb16-13"></a>The function used to calculate differential gene expression uses <span class="in">`HermesData`</span> as input.</span>
<span id="cb16-14"><a href="#cb16-14"></a>See <span class="co">[</span><span class="ot">RNAG1</span><span class="co">](../graphs/rnag01.qmd)</span> for details.</span>
<span id="cb16-15"><a href="#cb16-15"></a>In particular, the differential gene expression analysis does not require to normalize <span class="in">`HermesData`</span>.</span>
<span id="cb16-16"><a href="#cb16-16"></a></span>
<span id="cb16-17"><a href="#cb16-17"></a><span class="in">```{r, message = FALSE}</span></span>
<span id="cb16-18"><a href="#cb16-18"></a><span class="fu">library</span>(hermes)</span>
<span id="cb16-19"><a href="#cb16-19"></a></span>
<span id="cb16-20"><a href="#cb16-20"></a>object <span class="ot">&lt;-</span> hermes_data <span class="sc">%&gt;%</span></span>
<span id="cb16-21"><a href="#cb16-21"></a>  <span class="fu">add_quality_flags</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-22"><a href="#cb16-22"></a>  <span class="fu">filter</span>()</span>
<span id="cb16-23"><a href="#cb16-23"></a></span>
<span id="cb16-24"><a href="#cb16-24"></a><span class="fu">colData</span>(object) <span class="ot">&lt;-</span> <span class="fu">df_cols_to_factor</span>(<span class="fu">colData</span>(object))</span>
<span id="cb16-25"><a href="#cb16-25"></a><span class="in">```</span></span>
<span id="cb16-26"><a href="#cb16-26"></a></span>
<span id="cb16-27"><a href="#cb16-27"></a><span class="fu">## Differential Gene Expression Analysis via Limma-Voom Method</span></span>
<span id="cb16-28"><a href="#cb16-28"></a></span>
<span id="cb16-29"><a href="#cb16-29"></a>We can calculate a data frame in <span class="in">`HermesDataDiffExpr`</span> class to include the gene differential expression between groups in <span class="in">`HermesData`</span> using <span class="in">`diff_expression()`</span> function.</span>
<span id="cb16-30"><a href="#cb16-30"></a>By default, <span class="in">`diff_expression()`</span> function uses the <span class="in">`limma-voom`</span> method to calculate the gene differential expression, unless specified otherwise as argument.</span>
<span id="cb16-31"><a href="#cb16-31"></a>The <span class="in">`group`</span> argument is to set the groups for comparison.</span>
<span id="cb16-32"><a href="#cb16-32"></a>The <span class="in">`group`</span> factors should be a 2-level factor in <span class="in">`colData(HermesData)`</span>.</span>
<span id="cb16-33"><a href="#cb16-33"></a></span>
<span id="cb16-36"><a href="#cb16-36"></a><span class="in">```{r}</span></span>
<span id="cb16-37"><a href="#cb16-37"></a>result1 <span class="ot">&lt;-</span> <span class="fu">diff_expression</span>(object, <span class="at">group =</span> <span class="st">"SEX"</span>)</span>
<span id="cb16-38"><a href="#cb16-38"></a><span class="in">```</span></span>
<span id="cb16-39"><a href="#cb16-39"></a></span>
<span id="cb16-40"><a href="#cb16-40"></a><span class="fu">## Differential Gene Expression Analysis via DESeq2 Method</span></span>
<span id="cb16-41"><a href="#cb16-41"></a></span>
<span id="cb16-42"><a href="#cb16-42"></a>We can also use the <span class="in">`DESeq2`</span> Method for gene differential expression.</span>
<span id="cb16-43"><a href="#cb16-43"></a></span>
<span id="cb16-46"><a href="#cb16-46"></a><span class="in">```{r}</span></span>
<span id="cb16-47"><a href="#cb16-47"></a>result2 <span class="ot">&lt;-</span> <span class="fu">diff_expression</span>(object, <span class="at">group =</span> <span class="st">"SEX"</span>, <span class="at">method =</span> <span class="st">"deseq2"</span>)</span>
<span id="cb16-48"><a href="#cb16-48"></a><span class="in">```</span></span>
<span id="cb16-49"><a href="#cb16-49"></a></span>
<span id="cb16-50"><a href="#cb16-50"></a><span class="fu">## Additional Arguments</span></span>
<span id="cb16-51"><a href="#cb16-51"></a></span>
<span id="cb16-52"><a href="#cb16-52"></a>It is also possible to pass additional arguments to the function behind the methods in the <span class="in">`diff_expression()`</span>, ex.</span>
<span id="cb16-53"><a href="#cb16-53"></a>if we wish to change the type of size factor estimation when using the <span class="in">`DESeq2`</span> method.</span>
<span id="cb16-54"><a href="#cb16-54"></a>See <span class="in">`?DESeq2::DESeq()`</span> or <span class="in">`?limma::eBayes()`</span> for details about the additional parameters available.</span>
<span id="cb16-55"><a href="#cb16-55"></a></span>
<span id="cb16-58"><a href="#cb16-58"></a><span class="in">```{r}</span></span>
<span id="cb16-59"><a href="#cb16-59"></a>result3 <span class="ot">&lt;-</span> <span class="fu">diff_expression</span>(object, <span class="at">group =</span> <span class="st">"SEX"</span>, <span class="at">method =</span> <span class="st">"deseq2"</span>, <span class="at">sfType =</span> <span class="st">"ratio"</span>)</span>
<span id="cb16-60"><a href="#cb16-60"></a><span class="in">```</span></span>
<span id="cb16-61"><a href="#cb16-61"></a></span>
<span id="cb16-62"><a href="#cb16-62"></a><span class="fu">## Plot Differential Gene Expression in Volcano Graph</span></span>
<span id="cb16-63"><a href="#cb16-63"></a></span>
<span id="cb16-64"><a href="#cb16-64"></a>We can then plot a volcano graph for differential gene expression between groups in <span class="in">`HermesData`</span> using the <span class="in">`autoplot()`</span> function.</span>
<span id="cb16-65"><a href="#cb16-65"></a>The threshold values for adjusted p-values and log2 folder change can be specified.</span>
<span id="cb16-66"><a href="#cb16-66"></a></span>
<span id="cb16-69"><a href="#cb16-69"></a><span class="in">```{r}</span></span>
<span id="cb16-70"><a href="#cb16-70"></a><span class="fu">autoplot</span>(result1)</span>
<span id="cb16-71"><a href="#cb16-71"></a><span class="in">```</span></span>
<span id="cb16-72"><a href="#cb16-72"></a></span>
<span id="cb16-75"><a href="#cb16-75"></a><span class="in">```{r}</span></span>
<span id="cb16-76"><a href="#cb16-76"></a><span class="fu">autoplot</span>(result2, <span class="at">adj_p_val_thresh =</span> <span class="fl">0.025</span>, <span class="at">log2_fc_thresh =</span> <span class="dv">5</span>)</span>
<span id="cb16-77"><a href="#cb16-77"></a><span class="in">```</span></span>
<span id="cb16-78"><a href="#cb16-78"></a></span>
<span id="cb16-81"><a href="#cb16-81"></a><span class="in">```{r}</span></span>
<span id="cb16-82"><a href="#cb16-82"></a><span class="fu">autoplot</span>(result3)</span>
<span id="cb16-83"><a href="#cb16-83"></a><span class="in">```</span></span>
<span id="cb16-84"><a href="#cb16-84"></a></span>
<span id="cb16-85"><a href="#cb16-85"></a><span class="fu">## Teal Module for Sample Differential Gene Expression</span></span>
<span id="cb16-86"><a href="#cb16-86"></a></span>
<span id="cb16-87"><a href="#cb16-87"></a>We start by importing a <span class="in">`MultiAssayExperiment`</span>; here we use the example <span class="in">`multi_assay_experiment`</span> available in <span class="in">`hermes`</span>.</span>
<span id="cb16-88"><a href="#cb16-88"></a>It is wrapped as a <span class="in">`teal::dataset`</span>.</span>
<span id="cb16-89"><a href="#cb16-89"></a>We can then use the provided teal module <span class="in">`tm_g_volcanoplot`</span> to include a volcano graph module in our teal app.</span>
<span id="cb16-90"><a href="#cb16-90"></a></span>
<span id="cb16-91"><a href="#cb16-91"></a><span class="in">```{r,  message = FALSE, cache = FALSE}</span></span>
<span id="cb16-92"><a href="#cb16-92"></a><span class="co">#| screenshot.opts = list(delay = 24)</span></span>
<span id="cb16-93"><a href="#cb16-93"></a><span class="fu">library</span>(teal.modules.hermes)</span>
<span id="cb16-94"><a href="#cb16-94"></a>mae <span class="ot">&lt;-</span> hermes<span class="sc">::</span>multi_assay_experiment</span>
<span id="cb16-95"><a href="#cb16-95"></a>mae_data <span class="ot">&lt;-</span> <span class="fu">dataset</span>(<span class="st">"MAE"</span>, mae)</span>
<span id="cb16-96"><a href="#cb16-96"></a>data <span class="ot">&lt;-</span> <span class="fu">teal_data</span>(mae_data)</span>
<span id="cb16-97"><a href="#cb16-97"></a></span>
<span id="cb16-98"><a href="#cb16-98"></a>app <span class="ot">&lt;-</span> <span class="fu">init</span>(</span>
<span id="cb16-99"><a href="#cb16-99"></a>  <span class="at">data =</span> data,</span>
<span id="cb16-100"><a href="#cb16-100"></a>  <span class="at">modules =</span> <span class="fu">modules</span>(</span>
<span id="cb16-101"><a href="#cb16-101"></a>    <span class="fu">tm_g_volcanoplot</span>(</span>
<span id="cb16-102"><a href="#cb16-102"></a>      <span class="at">label =</span> <span class="st">"volcanoplot"</span>,</span>
<span id="cb16-103"><a href="#cb16-103"></a>      <span class="at">mae_name =</span> <span class="st">"MAE"</span></span>
<span id="cb16-104"><a href="#cb16-104"></a>    )</span>
<span id="cb16-105"><a href="#cb16-105"></a>  )</span>
<span id="cb16-106"><a href="#cb16-106"></a>)</span>
<span id="cb16-107"><a href="#cb16-107"></a><span class="fu">shinyApp</span>(app<span class="sc">$</span>ui, app<span class="sc">$</span>server)</span>
<span id="cb16-108"><a href="#cb16-108"></a><span class="in">```</span></span>
<span id="cb16-109"><a href="#cb16-109"></a></span>
<span id="cb16-110"><a href="#cb16-110"></a>{{&lt; include ../session_info.qmd &gt;}}</span>
<span id="cb16-111"><a href="#cb16-111"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">Made with â¤ï¸ by the Statistical Engineering Team</div>   
    <div class="nav-footer-center"><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>


<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>